### ISSUE TITLE
Get current consent-status to throw a more cleaner error in the logs

### ISSUE THREAD

**Issue by RamonQu** [skodaconnect/homeassistant-myskoda#686]
I saw quite some people with issues (#174, skodaconnect/homeassistant-skodaconnect#175, skodaconnect/homeassistant-skodaconnect#176, skodaconnect/homeassistant-skodaconnect#178) on the integration which is caused by a missing Consent in the MySkoda App.

I managed to capture the calls which are performed to GET the specific consent-status and POST a confirmation (by accepting the terms). Maybe it would be helpful to check the status of the consent and use it to throw a more cleaner error in the logging.

1. A GET call is done to https://api.connect.skoda-auto.cz/api/v1/users/[USERID]/identities which returns:
```
{
	"identities": [{
		"identity": "MBB",
		"actionsToCompleteIdentity": []
	}, {
		"identity": "DCS",
		"actionsToCompleteIdentity": []
	}, {
		"identity": "CONNECT",
		"actionsToCompleteIdentity": ["GIVE_DOCUMENT_CONSENT"]
	}, {
		"identity": "CABS",
		"actionsToCompleteIdentity": []
	}]
}
```
The GIVE_DOCUMENT_CONSENT is probably the status that prevents further actions to the API to get the car status.

2. A GET call is done to https://document-consent.vwgroup.io/api/v2/consents/required?document-id=[DOCID]&user-id=[USERID}&locale=en-GB

```
{
	"consentRequired": true,
	"versionDetails": {
		"id": "[ID]",
		"versionId": "[VERSIONID]",
		"locale": "en-GB",
		"title": "ŠKODA Connect",
		"text": "<br>\n\n### Terms and Conditions\n----------------\nImportant legal terms and information for setting up and using ŠKODA Connect services.\n\n[TERMS AND CONDITIONS ŠKODA CONNECT > ](https://consent.vwgroup.io/consent/v1/texts/SkodaConnect/gb/en/termsOfUse/latest/html)\n\n<br>\n\n### Information on Personal Data Processing\n----------------\nInformation on handling user data across ŠKODA Connect functions.\n\n[INFORMATION ON PERSONAL DATA PROCESSING > ](https://consent.vwgroup.io/consent/v1/texts/SkodaConnect/gb/en/dataprivacy/latest/html)",
		"disclaimer": ""
	},
	"consent": {
		"id": "[ID]",
		"userId": "[USERID]",
		"decision": true,
		"versionDetailsId": "[VERSIONID]",
		"collector": "[COLLECTOR]@apps_vw-dilab_com",
		"updatedAt": "2021-10-14T09:02:53Z",
		"version": 2
	}
}
```

3. A POST is done to https://document-consent.vwgroup.io/api/v2/consents to confirm the decision to accept the terms

Payload
{"decision":true,"userId":"[USERID]","versionDetailsId":"[VERSIONID]"}

Response
```
{
	"id": "[NEWID?]",
	"userId": "[USERID]",
	"decision": true,
	"versionDetailsId": "[OTHERVERSIONID],
	"collector": "[COLLECTOR]@apps_vw-dilab_com",
	"updatedAt": "2023-01-15T21:42:35.045Z",
	"version": 3
}
```

4. After accepting the terms the first GET call to https://api.connect.skoda-auto.cz/api/v1/users/[USERID]/identities return this response:

```
{
	"identities": [{
		"identity": "MBB",
		"actionsToCompleteIdentity": []
	}, {
		"identity": "DCS",
		"actionsToCompleteIdentity": []
	}, {
		"identity": "CONNECT",
		"actionsToCompleteIdentity": []
	}, {
		"identity": "CABS",
		"actionsToCompleteIdentity": []
	}]
}
```

**Comment by WebSpider** [comment]
It seems that a new URL was added to retrieve the current consent status in skodaconnect/myskoda#393

**Comment by WebSpider** [comment]
I am hesitant to blindly accept changed legal documents on behalf of the user.
We might want to see if we can raise an HA issue/problem with the new document, and then accept on behalf of the user when the user accepts this willfully.

**Comment by Simon-Spettmann** [comment]
> I am hesitant to blindly accept changed legal documents on behalf of the user. [...]

**What about adding option to auto-decline optional consent requests, e.g. marketing consent request?**

By doing this we wouldn't need to "blindly accept changed legal documents" on the users behalf yet still be able to reduce the number times the integrations stops working. We could also add a toggle in the integrations options and account setup flow that enables/disabled this behaviour.

I stumbled upon another project and a pull request [https://github.com/tillsteinbach/CarConnectivity-connector-skoda/pull/45](https://github.com/tillsteinbach/CarConnectivity-connector-skoda/pull/45), which seems to implement this.

**Comment by ds10git** [comment]
I've looked into the linked PR. It just extracts the link to continue the authentication. It will not accept the marketing options since it hasn't to be accepted to continue logging in.

I've also tried this PR for CarConnectivity and it's working, it just skips the marketing options and successfully returns the requested values while homeassistant-myskoda keeps displaying the marketing consent error.

**Comment by Simon-Spettmann** [comment]
I 've taken a close look at this and it seems to me that this should be patched in `skodaconnect/myskoda`. I also adopted the fix [https://github.com/tillsteinbach/CarConnectivity-connector-skoda/pull/45](https://github.com/tillsteinbach/CarConnectivity-connector-skoda/pull/45) (with the help of AI), see: https://github.com/Simon-Spettmann/myskoda/commit/d09ea4d3d587da4e375121711a46caeffa2fe131. As soon as I get another marketing consent request I'll test it myself.

@WebSpider / @dvx76 Would you be open to a PR in `skodaconnect/myskoda`?

**Comment by WebSpider** [comment]
Yes, this should be patched there and we would be open to a PR there. 👍
